index.js

```diff
@@ -103,6 +72,43 @@ function getScmUriParts(scmUri) {
 }
 
 class GitlabScm extends Scm {
+    /**
+     * Github command to run
+     * @method _gitlabCommand
+     * @param  {Object}      options              An object that tells what command & params to run
+     * @param  {String}      options.action       Github method. For example: get
+     * @param  {String}      options.token        Github token used for authentication of requests
+     * @param  {Object}      options.params       Parameters to run with
+     * @param  {String}      [options.scopeType]  Type of request to make. Default is 'repos'
+     * @param  {String}      [options.route]      Route for gitlab.request()
+     * @param  {Function}    callback             Callback function from gitlab API
+     */
+    _gitlabCommand(options, callback) {
+        const config = {
+            url: `${this.gitlabConfig.prefixUrl}/${options.route}`, // Generate url
+            method: options.method
+        };
+
+        delete options.method;
+        delete options.route;
+
+        // Json goes at top level of config
+        if (options.json) {
+            config.json = options.json;
+            delete options.json;
+        }
+
+        // Everything else goes into context
+        config.context = options;
+
+        got(config)
```

Comment:
can't we use async await here ?  Right now`_gitlabCommand` will exit immediately, looks like we are doing it this say for force callback functions for breaker